webpack.common.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+const path = require('path');
+const url = require('url');
+const nconf = require('nconf');
+
+const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+
+const activePlugins = require('./build/active_plugins.json');
+
+let relativePath = nconf.get('relative_path');
+if (relativePath === undefined) {
+	nconf.file({
+		file: path.resolve(__dirname, nconf.any(['config', 'CONFIG']) || 'config.json'),
+	});
+
+	const urlObject = url.parse(nconf.get('url'));
+	relativePath = urlObject.pathname !== '/' ? urlObject.pathname.replace(/\/+$/, '') : '';
+}
+
+module.exports = {
+	plugins: [
+		new CleanWebpackPlugin(), // cleans dist folder
+	],
+	entry: {
+		nodebb: './build/public/src/client.js',
+		admin: './build/public/src/admin/admin.js',
+	},
+	output: {
+		filename: '[name].min.js',
```

Comment:
Would recommend adding a hash to the filename for cache purposes.

Reply:
https://github.com/NodeBB/NodeBB/commit/e190ff875acf97a6eaa513171ff78819fb4b4cb0